stdlib/public/core/Integers.swift.gyb

```diff
@@ -1626,22 +1626,33 @@ extension BinaryInteger {
   // FIXME(ABI): using Int as the return type is wrong.
   @_transparent
   public func distance(to other: Self) -> Int {
-    let distance = other - self
-    if let result = Int(exactly: distance) {
-      return result
+    let isNegative = Self.isSigned && self < 0
+    let sameSign = !Self.isSigned || (isNegative == (other < 0))
+
+    if sameSign {
+      if self.magnitude > other.magnitude {
```

Comment:
It looks like one can simplify the code a bit.

Say we calculate `delta = self.magnitude - other.magnitude`.

example | delta | delta < 0 | isNegative | finalResult
-- | -- | -- | -- | --
2.distance(to: 5) | -3 | true | false | 3 (== `-delta`)
5.distance(to: 2) | 3 | false | false | -3 (== `-delta`)
(-2).distance(to: -5) | -3 | true | true | -3 (== 'delta')
(-5).distance(to: -2) | 3 | false | true | 3 (== `delta`)

So it looks like this should work..

```swift
if sameSign {
  let delta = self.magnitude - other.magnitude
  if let result = Int(exactly: delta) {
    return isNegative ? result : -result
  }
} else {
  // this part is unchanged
  if let result = Int(exactly: self.magnitude + other.magnitude) {
    return isNegative ? result : -result
  }
}
```

which can be simplified a little more:

```swift
let distance = sameSign
  ? (self.magnitude - other.magnitude)
  : (self.magnitude + other.magnitude)
if let result = Int(exactly: distance) {
  return isNegative ? result : -result
}
```

If this is correct (which I'm not 💯 sure), this should probably lead to a better code. I would be interested in looking at the SIL and IR being generated.

Reply:
Oh... I take that back, the type can be unsigned and then the negative delta will not be representable ;-(

Reply:
Right… Between signed types not being able to represent their whole range and unsigned types not being able to represent negative values, this is kind of an annoying method.

Reply:
@natecook1000 I *think* there may be a slightly more simple way to do this that may also be a little more performant. Won't have time to try until Saturday, but I will update if it works out :) Anyway, glad that this is getting some attention.